src/Language/Source.php

```diff
@@ -28,7 +33,8 @@ public function __construct($body, $name = null)
         );
 
         $this->body = $body;
-        $this->length = mb_strlen($body, 'UTF-8');
+        $this->buf =  array_map('GraphQL\Utils::ord', preg_split('//u', $body, -1, PREG_SPLIT_NO_EMPTY));
```

Comment:
@vladar I have a naive implementation using the original `$body` string as a packed value. I have the tests passing on my Windows environment, but I'm getting segfaults in my CentOS VM (this might be related to some other customizations going on). That said, I don't have a real stream implementation, but given how inefficient the base PHP value types are, parsing as a C module would certainly be much better.